033-OSSDatabaseMigration/Coach/05-online-migration.md

```diff
@@ -0,0 +1,223 @@
+# Challenge 5: Online migration
+
+[< Previous Challenge](./04-offline-cutover-validation.md) - **[Home](./README.md)** - [Next Challenge >](./06-online-cutover-validation.md)
+
+## Coach Tips
+
+The intent of this WTH is not to focus on networking so if the attendee has trouble configuring VPN/VNET peering, feel free to hand hold them through it
+Remind the attendees to review the hints section if they get stuck
+
+## Introduction
+
+Perform an online migration using the Azure Database Migration Service
+
+## Steps -- PostgreSQL
+
+Connect to database container and run the export
+
+```bash
+kubectl -n postgresql exec deploy/postgres -it -- bash
+
+pg_dump -o -h localhost -U contosoapp -d wth -s >dump_wth.sql
+```
+
+This creates a psql dump text file. We need to import it to the target - schema only. Create a separate database for online migration (suggeted ). Alternately you can drop and re-create the database wth.
+
+To drop all the tables with indexes
+
+```bash
+\out drop_tables.sql
+
+select 'drop table ' || tablename || ' cascade;'  from pg_tables where tableowner = 'contosoapp' and schemaname = 'public' ;
+```
+
+To import the schema only to target using psql
+
+```bash
+$ psql -h pgtarget.postgres.database.azure.com -p 5432 -U contosoapp@pgtarget  -d wth <dump_wth.sql
+
+postgres=> create database wth
+```
+
+Import the schema to target
+
+```bash
+psql -h pgtarget.postgres.database.azure.com -U contosoapp@pgtarget -d wth2 < dump_wth.sql
+```
+
+Verify count of tables and indexes created on the target database from psql. 26tables, 39 indices.
+
+```bash
+\dt+
+
+\di+
+
+```
+
+The query to disable all foreign key in DMS  (https://docs.microsoft.com/en-us/azure/dms/tutorial-postgresql-azure-postgresql-online-portal )  shows more results than needed. It can be simplified bu saving the output of a SQL command that just drops the constraints. From psql,
```

Comment:
Hyperlink the URL to hide it OR wrap it in angle brackets so that it is proper markdown

Reply:
Fixed